### PR TITLE
GUI-2551, prevent expression interpolation in values returned from api.

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2551

### CODE REVIEW

File: eucaconsole/static/js/pages/alarms/alarm_detail.js [hunk 1]
```diff
@@ -5,27 +5,35 @@ angular.module('AlarmDetailPage', [
...
-        link: function (scope, element, attrs) {
-            scope.alarm = JSON.parse(attrs.alarmDetail);
-            scope.alarm.actions = scope.alarm.actions || [];
-            scope.alarms = [scope.alarm];  // Delete alarm confirmation dialog expects a list of alarms
-            scope.expanded = true;
-            scope.alarmDimensions = scope.alarm.dimensions;  // Leveraged in delete alarm confirmation dialog
-            // Need stringified form on details page (and Copy Alarm dialog) to set current dimension choice
-            scope.alarm.dimensions = JSON.stringify(scope.alarm.dimensions);
+        compile: function (tElement, tAttrs) {
+            var alarm = JSON.parse(tAttrs.alarmDetail);
```
Comment (coyote240) [inline eucaconsole/static/js/pages/alarms/alarm_detail.js]: Seems that deserializing the alarm details before the link function preempts expression interpolation.

File: eucaconsole/static/js/pages/alarms/alarm_detail.js [hunk 2]
```diff
@@ -52,23 +60,6 @@ angular.module('AlarmDetailPage', [
                     });
             };
 
-            $scope.$on('actionsUpdated', function (event, actions) {
```
Comment (coyote240) [inline eucaconsole/static/js/pages/alarms/alarm_detail.js]: This seems to have come over in a cut'n'paste.  Actions are collated in a watch below.

File: eucaconsole/static/js/pages/alarms/alarm_detail.js (hunk 1 above)
Comment (dkavanagh) [inline eucaconsole/static/js/pages/alarms/alarm_detail.js]: So, the scope.alarm value gets bound to the UI field, but not evaluated? I believe what you're saying, but wanted to understand what's happening more.

Comment (coyote240) [inline eucaconsole/static/js/pages/alarms/alarm_detail.js]: As I understand it.  Now, notice that it's not scope.alarm that is bound to the template but scope.alarm.description.  It's getting pretty deep into angularjs internals that aren't documented super-great, but it seems that by deserializing and assigning to scope.alarm before linking we get ahead of expression evaluation.  I wouldn't do this in too many places, but here there is certainly no desire to allow expressions to be stored or evaluated.

**Pull request merged**